lib/hashes.md

```diff
@@ -1,44 +1,33 @@
 ---
 title: Hashes
 instructor_notes: Feel free to re-organize the headings (or add/remove headings) below. We included the headings for your benefit, but it's 100% fine if you want to write your responses in some different structure.
----
-
 # What is a Hash?
 
-Replace me with your response...
-
-# What are some examples of information that would be Hashes as opposed to some other data type?
+A hash is a list of things (keys) and data associated with those things (values).
 
-Replace me with your response...
+(# What are some examples of information that would be Hashes as opposed to some other data type? & # How are Hashes and Arrays similar? How are they different?)
 
-# How are Hashes and Arrays similar? How are they different?
-
-Replace me with your response...
+Hashes and arrays are both lists that store data, and one can perform functions to each of the items on the list. However, hashes allow you add specific values to the different items on the list (prices on menu items, ages of students, or various facts about a single user [age=25, height=6’4, etc]) They are also organized very differently. An array has numbered slots, so you have to remember the number of the slot in order to retrieve data, and new items are added to either the beginning or end of the array (and thus they get their own numbered slot). In a Hash, it’s more like…a pile. Things just get added to the pile, with no beginning and no end.
 
 # How do you retrieve a particular value from a Hash?
+To retrieve a value, you just need the name of the hash and the name of the key whose value you are trying to find. Example foods[:pizza] would return => “best”
+(Shannon question: this doesn’t work the opposite way. How would I search for “best” on a hash of foods?)
```

Comment:
Typically, you wouldn't. Hashes are meant to be used from the perspective of the key–not the value. That said, it's possible to search in the reverse if you build that functionality yourself. Ruby just doesn't give us a neat one-liner to do it.